helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -52,16 +56,26 @@
   private static final String ACCEPT_CONTENT_TYPE = "application/json";
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private HttpClient _httpClient;
+  private final HttpClient _httpClient;
+  private final Semaphore _semaphore;
+  // a transient cache holds the HttpResponse of HttpPosts, only valid when cache is enabled
+  private final Map<Integer, HttpResponse> _cachedResults = new HashMap<>();
```

Comment:
Do we ever clear this cache?

Reply:
No, we don't need to, a new instance of CustomRestClientImpl instead of singleton will be created per each request. That's the idea of keeping rest service stateless

Reply:
I took a second look and you are correct. Just getting used to GitHub's code review layout. I'll resolve this comment.